docker_auth_test.go

```diff
@@ -0,0 +1,104 @@
+package testcontainers
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const exampleAuth = "https://example-auth.com"
+const indexDockerIO = "https://index.docker.io/v1/"
+
+var testDockerConfigDirPath = filepath.Join("testresources", ".docker")
+
+func TestGetDockerConfig(t *testing.T) {
+	t.Run("without DOCKER_CONFIG env var retrieves default", func(t *testing.T) {
+		cfg, err := getDockerConfig()
+		require.Nil(t, err)
+		require.NotNil(t, cfg)
+
+		assert.Equal(t, 1, len(cfg.AuthConfigs))
+
+		authCfgs := cfg.AuthConfigs
+
+		if _, ok := authCfgs[indexDockerIO]; !ok {
+			t.Errorf("Expected to find %s in auth configs", indexDockerIO)
+		}
+	})
+
+	t.Run("with DOCKER_CONFIG env var pointing to a non-existing file raises error", func(t *testing.T) {
+		t.Setenv("DOCKER_CONFIG", filepath.Join(testDockerConfigDirPath, "non-existing"))
```

Comment:
I assume `Setenv` is only persisted in the context of the test and does not pollute into the whole process for other tests, right?

Reply:
Exactly that, testing library will handle it